### PR TITLE
Remove CVO service monitor

### PR DESCRIPTION
**Pull request by paulfantom**
Fixes MON-675

/cc @brancz 

Hold until we apply Service Monitor in CVO repo (https://github.com/openshift/cluster-version-operator/pull/214).
/hold

### CODE REVIEW

**Comment by paulfantom** [comment]
That test is ok as it is checking if we are scraping data from cluster-version-operator. We are only moving ServiceMonitor definition to CVO repo (it is already there: https://github.com/openshift/cluster-version-operator/blob/master/install/0000_90_cluster-version-operator_02_servicemonitor.yaml), but we still should be testing if that SM is applied.

I started a cluster and it seems that metrics from CVO are still not getting into prometheus.

File: jsonnet/prometheus.jsonnet [hunk 1]
```diff
@@ -114,35 +114,7 @@ local namespacesRole =
 
     // OpenShift has the kube-apiserver as well as an aggregated API called
```
Comment (LiliC) [inline jsonnet/prometheus.jsonnet]: Is this comment still relevant?

Comment (paulfantom) [reply jsonnet/prometheus.jsonnet]: No, I will remove it in https://github.com/openshift/cluster-monitoring-operator/pull/438

**Pull request merged**